osu.Game/Online/Chat/DrawableLinkCompiler.cs

```diff
@@ -52,6 +77,45 @@ private void load(OsuColour colours)
 
         protected override IEnumerable<Drawable> EffectTargets => Parts;
 
+        public MenuItem[] ContextMenuItems
+        {
+            get
+            {
+                List<MenuItem> items = new List<MenuItem>();
+
+                if (link is not null)
+                {
+                    string text;
+
+                    switch (link.Action)
+                    {
+                        case LinkAction.OpenChannel:
+                        case LinkAction.JoinMultiplayerMatch:
+                            text = "Join";
+                            break;
+
+                        case LinkAction.External:
+                            text = "Visit";
+                            break;
+
+                        default:
+                            text = "Open";
+                            break;
+                    }
+
+                    items.Add(new OsuMenuItem(text, MenuItemType.Highlighted, () => linkHandler?.HandleLink(link.Url)));
+
+                    items.Add(new OsuMenuItem("Copy URL", MenuItemType.Standard, () =>
```

Comment:
I would make user links work (in beatmap cards, comments) or is it out of scope of this PR?